src/CommunityVoices/Model/Mapper/LocationCollection.php

```diff
@@ -0,0 +1,27 @@
+<?php
+
+namespace CommunityVoices\Model\Mapper;
+
+use PDO;
+use CommunityVoices\Model\Component\DataMapper;
+use CommunityVoices\Model\Entity;
+
+class LocationCollection extends DataMapper
+{
+    public function fetch(Entity\LocationCollection $container)
+    {
+        $this->fetchAll($container);
+    }
+
+    public function fetchAll(Entity\LocationCollection $container)
+    {
+        $query = "SELECT id, label, end_use FROM `community-voices_locations` ORDER BY end_use DESC, label ASC";
+        foreach ($this->conn->query($query) as $row) {
```

Comment:
can you use `addEntityFromParams`? mappers do not need express knowledge of how to create domain objects

Reply:
Does https://github.com/EnvironmentalDashboard/community-voices/pull/19/commits/532e5b7e29d28ad98af83787c97ec8ffbeddfa8f cover this entirely?  Is it still fine to do the `$this->conn->query($query)` in the `foreach`?  The old code that I referenced had `prepare` called more often, so I don't know if this is good practice still.

Reply:
did you test this? I did not, but I figure it won't work. look at how other SELECT queries use 'AS'. i figure it's OK to leave it in the for()

Reply:
The query itself is the same one used historically - I copied that line from the old code.  So it both works here and also has worked for history.

Reply:
`end_use` does actually line up correctly.  I did add code to make sure that ID is an integer, but all looks good here.

Reply:
oh. i see. php method names are case insensitive.